sources/metricdog/src/main.rs

```diff
@@ -113,7 +124,7 @@ pub(crate) fn main_inner(arguments: Arguments, service_check: Box<dyn ServiceChe
     .context(error::BottlerocketReleaseSnafu)?;
 
     // instantiate the metricdog object
-    let metricdog = Metricdog::from_parts(config, os_release, service_check)?;
+    let metricdog = Metricdog::from_parts(config, os_release, service_check, host_check)?;
```

Comment:
4 may already be too many arguments. What about a `MetricDogParams` struct (or a different name if you have a better proposal)?. I've seen this is a common practice in other crates (e.g., `rustls`).

Reply:
I don't think adding a struct definition for the sake of being able to pass a single object to a constructor would enhance with clarity here. I'm willing to be convinced otherwise.